sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -158,12 +158,28 @@ class FileStreamSource(
   }
 
   /**
+   * If the source has a metadata log indicating which files should be read, then we should use it.
+   * We figure out whether there exists some metadata log only when user gives a non-glob path.
+   */
+  private val sourceHasMetadata: Boolean =
```

Comment:
Actually, why not just change `sourceHasMetadata` to a method? `sparkSession.sessionState.newHadoopConf()` seems expensive but we can save it into a field.